File: BackgroundChangerAndroid/src/com/jbutewicz/backgroundchanger/ColorGenerator.java
package com.jbutewicz.backgroundchanger;

import java.util.ArrayList;
import java.util.Random;

public class ColorGenerator {

	public static ArrayList<Integer> generateNewColor() {
		Random generator = new Random(); 
		Integer red = generator.nextInt(256);
		Integer green = generator.nextInt(256);
		Integer blue = generator.nextInt(256);
		ArrayList<Integer> color = new ArrayList<Integer>();
		color.add(red);
		color.add(green);	
		color.add(blue);
		return color;
		
	}

	public static void pushColor(ArrayList<ArrayList<Integer>> colors) {
		colors.add(generateNewColor());
	}
	
	public static ArrayList<Integer> popColor(ArrayList<ArrayList<Integer>> colorArray) {
		ArrayList<Integer> color = new ArrayList<Integer>();
		color = colorArray.get(0);
		colorArray.remove(0);
		return color;
	
	}
	
}

File: BackgroundChangerIOS/BackgroundChanger/ColorGenerator.m
//
//  ColorGenerator.m
//  BackgroundChanger
//
//  Created by Joseph Butewicz on 9/4/14.
//  Copyright (c) 2014 jbutewicz. All rights reserved.
//

#import "ColorGenerator.h"

@interface ColorGenerator()

@end

@implementation ColorGenerator

- (NSMutableArray *)generateNewColor {
    NSNumber *red = [NSNumber numberWithInt:arc4random_uniform(256)];
    NSNumber *green = [NSNumber numberWithInt:arc4random_uniform(256)];
    NSNumber *blue = [NSNumber numberWithInt:arc4random_uniform(256)];
    NSMutableArray *color = [[NSMutableArray alloc] init];
    [color addObject: red];
    [color addObject: green];
    [color addObject: blue];
    return color;
}

- (void)pushColor:(NSMutableArray *)colorArray {
    [colorArray addObject:[self generateNewColor]];
}

- (NSMutableArray *)popColor:(NSMutableArray *)colorArray {
    NSMutableArray *color = [[NSMutableArray alloc] init];
    color = [colorArray objectAtIndex:0];
    [colorArray removeObjectAtIndex:0];
    return color;
}

@end